infrastructure_files/getting-started-with-zitadel.sh

```diff
@@ -402,29 +402,126 @@ read_nb_domain() {
+  read -r READ_NETBIRD_PORT < /dev/tty
+  if ! check_nb_port "$READ_NETBIRD_PORT"; then
+    read_nb_port
+  fi
+  echo "$READ_NETBIRD_PORT"
+}
+
+
+check_nb_http_port() {
+  HTTP_PORT=$1
+  if [ "$HTTP_PORT-x" == "-x" ]; then
+    echo "The NETBIRD_HTTP_PORT variable cannot be empty." > /dev/stderr
+    return 1
+  fi
+  return 0
+}
+
+read_nb_http_port() {
+  READ_NETBIRD_HTTP_PORT=""
+  echo -n "Enter the http port you want to use for NetBird (e.g. 80): " > /dev/stderr
+  read -r READ_NETBIRD_HTTP_PORT < /dev/tty
+  if ! check_nb_http_port "$READ_NETBIRD_HTTP_PORT"; then
+    read_nb_http_port
+  fi
+  echo "$READ_NETBIRD_HTTP_PORT"
+}
+
+check_nb_8080_port() {
+  PORT_8080=$1
+  if [ "$PORT_8080-x" == "-x" ]; then
+    echo "The NETBIRD_8080_PORT variable cannot be empty." > /dev/stderr
+    return 1
+  fi
+  return 0
+}
+
+read_nb_8080_port() {
+  READ_NETBIRD_8080_PORT=""
+  echo -n "Enter the NETBIRD_8080_PORT you want to use for NetBird (e.g. 8080): " > /dev/stderr
+  read -r READ_NETBIRD_8080_PORT < /dev/tty
+  if ! check_nb_8080_port "$READ_NETBIRD_8080_PORT"; then
+    read_nb_8080_port
+  fi
+  echo "$READ_NETBIRD_8080_PORT"
+}
+
+
+check_nb_3478_port() {
+  PORT_3478=$1
+  if [ "$PORT_3478-x" == "-x" ]; then
+    echo "The NETBIRD_3478_PORT variable cannot be empty." > /dev/stderr
+    return 1
+  fi
+  return 0
+}
+
+read_nb_3478_port() {
+  READ_NETBIRD_3478_PORT=""
+  echo -n "Enter the TURN_LISTENING_PORT you want to use for NetBird (e.g. 3478): " > /dev/stderr
+  read -r READ_NETBIRD_3478_PORT < /dev/tty
+  if ! check_nb_3478_port "$READ_NETBIRD_3478_PORT"; then
+    read_nb_3478_port
+  fi
+  echo "$READ_NETBIRD_3478_PORT"
+}
+
 initEnvironment() {
   CADDY_SECURE_DOMAIN=""
   ZITADEL_EXTERNALSECURE="false"
   ZITADEL_TLS_MODE="disabled"
   ZITADEL_MASTERKEY="$(openssl rand -base64 32 | head -c 32)"
-  NETBIRD_PORT=80
+  #NETBIRD_PORT=80
   NETBIRD_HTTP_PROTOCOL="http"
   TURN_USER="self"
   TURN_PASSWORD=$(openssl rand -base64 32 | sed 's/=//g')
   TURN_MIN_PORT=49152
   TURN_MAX_PORT=65535
+  #TURN_LISTENING_PORT=3478
 
   if ! check_nb_domain "$NETBIRD_DOMAIN"; then
     NETBIRD_DOMAIN=$(read_nb_domain)
   fi
 
+  if ! check_nb_http_port "$NETBIRD_HTTP_PORT"; then
+    NETBIRD_HTTP_PORT=$(read_nb_http_port)
+  fi
+
+  if ! check_nb_http_port "$NETBIRD_8080_PORT"; then
+    NETBIRD_8080_PORT=$(read_nb_8080_port)
+  fi
+
+  if ! check_nb_3478_port "$TURN_LISTENING_PORT"; then
+    TURN_LISTENING_PORT=$(read_nb_3478_port)
+  fi
+
   if [ "$NETBIRD_DOMAIN" == "use-ip" ]; then
     NETBIRD_DOMAIN=$(get_main_ip_address)
+	  NETBIRD_PORT=$NETBIRD_HTTP_PORT
   else
     ZITADEL_EXTERNALSECURE="true"
     ZITADEL_TLS_MODE="external"
-    NETBIRD_PORT=443
-    CADDY_SECURE_DOMAIN=", $NETBIRD_DOMAIN:$NETBIRD_PORT"
+    #NETBIRD_PORT=443
+
+    if ! check_nb_port "$NETBIRD_PORT"; then
+      NETBIRD_PORT=$(read_nb_port)
+    fi
+
+    CADDY_SECURE_DOMAIN=", $NETBIRD_DOMAIN:443"
```

Comment:
Is the hardcoded 443 port correct? Maybe the proper port is the NETBIRD_PORT variable.